### DISCUSSION TITLE
DS18Alias with alphanumeric alias does not work

### DISCUSSION THREAD

**Discussion by infa-presales** [arendst/Tasmota#20139] [Q&A]
Hi, 
I'm using Tasmota 13.2.0.2 and use a self-compiled version on a nodeMCU board for DS18x20 monitor.
I have added 
```
#ifndef USE_RULES
#define USE_RULES
#endif

#ifndef DS18x20_USE_ID_ALIAS
#define DS18x20_USE_ID_ALIAS                   // Add support aliasing for DS18x20 sensors. See comments in xsns_05 files (+0k5 code)
#endif

#ifndef DS18x20_USE_ID_AS_NAME
#define DS18x20_USE_ID_AS_NAME                 // Use last 3 bytes for naming of sensors
#endif
```
and as a result with my test sensor (just single sensor to see if it works) I see in Main Menu my sensor nicely show with the 3 byte name
DS18B20-70C9EE Temperatur  21,5 °C
The DS18Alias works and yield the 64bit Id

```
16:28:34.355 CMD: DS18alias
16:28:34.360 RSL: RESULT = {"DS18B20-70C9EE":{"Id":"7C02162270C9EE28"}}
```
Then I set
```
Rule1 ON System#Boot DO
Backlog DS18Alias 7C02162270C9EE28,HK1VL;
ENDON
```
and activated the Rule1 with 
```
Rule1 1 
Rule1
```
Here the log
```
16:30:53.590 CMD: Rule1 ON System#Boot DO Backlog DS18Alias 7C02162270C9EE28,HK1VL; ENDON
16:30:53.594 RUL: Stored uncompressed, would compress from 65 to 60 (-8%)
16:30:53.596 RSL: RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":65,"Free":446,"Rules":"ON System#Boot DO Backlog DS18Alias 7C02162270C9EE28,HK1VL; ENDON"}}
16:31:00.240 CMD: Rule1 1
16:31:00.244 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":65,"Free":446,"Rules":"ON System#Boot DO Backlog DS18Alias 7C02162270C9EE28,HK1VL; ENDON"}}
16:31:01.277 MQT: Verbindungsversuch…
16:31:01.484 MQT: Verbindung fehlgeschlagen aufgrund von 192.168.2.177:1883, rc -2. Wiederversuch in 120 s
16:31:14.851 CMD: Rule1
16:31:14.855 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":65,"Free":446,"Rules":"ON System#Boot DO Backlog DS18Alias 7C02162270C9EE28,HK1VL; ENDON"}}
```
However that does not change neither the shipping nor the display. Even when I try to assign an alias with 
```
DS18Alias 7C02162270C9EE28,HK1VL
```
(I'm aware it won't survive a reboot/restart)  it seem not to work:
```
16:43:42.814 CMD: DS18Alias 7C02162270C9EE28,HK1VL
16:43:42.819 RSL: RESULT = {"DS18B20-70C9EE":{"Id":"7C02162270C9EE28"}}
16:44:44.382 RSL: STATE = {"Time":"2023-12-01T16:44:44","Uptime":"0T00:35:04","UptimeSec":2104,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":24,"MqttCount":0,"Wifi":{"AP":2,"SSId":"WLAN-AABB6B","BSSId":"C0:25:06:99:10:AC","Channel":1,"Mode":"11n","RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:44:44.387 RSL: SENSOR = {"Time":"2023-12-01T16:44:44","DS18B20-70C9EE":{"Id":"02162270C9EE","Temperature":21.3},"TempUnit":"C"}
```
SENSOR Events and the display did not change and are not using the alphanumeric alias. Same result if I try numeric values
```
16:47:08.266 CMD: DS18Alias 7C02162270C9EE28,1
16:47:08.271 RSL: RESULT = {"DS18B20-70C9EE":{"Id":"7C02162270C9EE28"}}
```
Do I miss something here (e.g. that it works only with more then one sensor)?

I also compiled the tasmota without DS18x20_USE_ID_ALIAS to see what it changes and then the DS18alias command is not supported anymore, so I think that the flag is active. Same if i disable DS18x20_USE_ID_AS_NAME I just get the "Temparature" alias only. 
[ I will have to measure in the floor heating systems upto 14 sensors and like to have more speaking names then just hexcodes.]
Thanks 
    Detlef

**Comment by infa-presales** [comment]
Ah, I guess I can aswer this question myself: After DISABLE  DS18x20_USE_ID_AS_NAME  it seems to work. So it looks as if the DS18Alias only partially working then (command works, but setting of aliases does not as they are overwritten.
```
17:02:46.901 CMD: DS18Alias 7C02162270C9EE28,HK1VL
17:02:46.905 RSL: RESULT = {"HK1VL":{"Id":"7C02162270C9EE28"}}
```

**Reply by sfromis** [reply]
Right. You can't have both.

**Comment by infa-presales** [comment]
Yes, I checked source, which is not flexible due to use of defines. It would be preferable if it was different, but I have to agree that we can live with the current implementation, just need a dedicated custom compile. I close this discussion.
